.github/workflows/dev.yml

```diff
@@ -0,0 +1,35 @@
+# This workflow will build a Java project with Maven
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: Java CI with Maven
+
+on:
+  push:
+    branches: [ dev ]
+  pull_request:
+    branches: [ dev ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 1.11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.11
+      - name: Build with Maven
+        run: mvn clean compile install -DskipTests --file pom.xml
```

Comment:
I don't think we need separate tests running from the main build at this point of time. There is no reason for it. Also In the current configuration, you perform a bunch of maven lifecycle steps twice. Let's keep it simple for now.